# Author:       Toxikuu
# Timestamp:    2025-04-12 15:02
# Version:      0.21.0
# Package:      eza
# License:      Upstream's
# Description:  Tweak and add icons to eza

diff --git a/build.rs b/build.rs
index 201d495f..c3e17457 100644
--- a/build.rs
+++ b/build.rs
@@ -131,6 +131,8 @@ fn nonstandard_features_string() -> String {
         s.push("-git");
     }

+    s.push("+tox_icons_patch");
+
     s.join(", ")
 }

diff --git a/src/output/icons.rs b/src/output/icons.rs
index 27336efb..98d54fac 100644
--- a/src/output/icons.rs
+++ b/src/output/icons.rs
@@ -14,6 +14,7 @@ struct Icons;
 
 #[rustfmt::skip]
 impl Icons {
+    const ARCH_LINUX: char      = '\u{f303}';  // 
     const AUDIO: char           = '\u{f001}';  // 
     const BINARY: char          = '\u{eae8}';  // 
     const BOOK: char            = '\u{e28b}';  // 
@@ -21,8 +22,9 @@ impl Icons {
     const CACHE: char           = '\u{f49b}';  // 
     const CAD: char             = '\u{f0eeb}'; // 󰻫
     const CLOCK: char           = '\u{f43a}';  // 
+    const CMAKE: char           = '\u{e794}';  // 
     const COMPRESSED: char      = '\u{f410}';  // 
-    const CONFIG: char          = '\u{e615}';  // 
+    const CONFIG: char          = '\u{f107b}'; // 󱁻
     const CSS3: char            = '\u{e749}';  // 
     const DATABASE: char        = '\u{f1c0}';  // 
     const DIFF: char            = '\u{f440}';  // 
@@ -36,8 +38,10 @@ impl Icons {
     const ESLINT: char          = '\u{e655}';  // 
     const FILE: char            = '\u{f15b}';  // 
     const FILE_3D: char         = '\u{f01a7}'; // 󰆧
+    const FILE_LOCK: char       = '\u{f0221}'; // 󰈡
     const FILE_OUTLINE: char    = '\u{f016}';  // 
     const FOLDER: char          = '\u{e5ff}';  // 
+    const FOLDER_CODE: char     = '\u{f0d09}'; // 󰴉
     const FOLDER_CONFIG: char   = '\u{e5fc}';  // 
     const FOLDER_EXERCISM: char = '\u{ebe5}';  // 
     const FOLDER_GIT: char      = '\u{e5fb}';  // 
@@ -47,11 +51,12 @@ impl Icons {
     const FOLDER_NPM: char      = '\u{e5fa}';  // 
     const FOLDER_OCAML: char    = '\u{e67a}';  // 
     const FOLDER_OPEN: char     = '\u{f115}';  // 
+    const FOLDER_WRENCH: char   = '\u{f19fc}'; // 󱧼
     const FONT: char            = '\u{f031}';  // 
     const FREECAD: char         = '\u{f336}';  // 
     const GIMP: char            = '\u{f338}';  // 
     const GIST_SECRET: char     = '\u{eafa}';  // 
-    const GIT: char             = '\u{f1d3}';  // 
+    const GIT: char             = '\u{f02a2}'; // 󰊢
     const GODOT: char           = '\u{e65f}';  // 
     const GRADLE: char          = '\u{e660}';  // 
     const GRAPH: char           = '\u{f1049}'; // 󱁉
@@ -105,12 +110,13 @@ impl Icons {
     const LANG_TYPESCRIPT: char = '\u{e628}';  // 
     const LANG_V: char          = '\u{e6ac}';  // 
     const LIBRARY: char         = '\u{eb9c}';  // 
-    const LICENSE: char         = '\u{f02d}';  // 
+    const LICENSE: char         = '\u{f02d}';  // 
     const LOCK: char            = '\u{f023}';  // 
     const LOG: char             = '\u{f18d}';  // 
     const MAKE: char            = '\u{e673}';  // 
     const MARKDOWN: char        = '\u{f48a}';  // 
     const MUSTACHE: char        = '\u{e60f}';  // 
+    const NEWS: char            = '\u{f1ea}';  // 
     const NODEJS: char          = '\u{e718}';  // 
     const NPM: char             = '\u{e71e}';  // 
     const OS_ANDROID: char      = '\u{e70e}';  // 
@@ -135,13 +141,16 @@ impl Icons {
     const SIGNED_FILE: char     = '\u{f19c3}'; // 󱧃
     const SLIDE: char           = '\u{f1c4}';  // 
     const SQLITE: char          = '\u{e7c4}';  // 
+    const STEAM: char           = '\u{f1b6}';  // 
     const SUBLIME: char         = '\u{e7aa}';  // 
     const SUBTITLE: char        = '\u{f0a16}'; // 󰨖
+    const TCL: char             = '\u{f06d3}'; // 󰛓
     const TERRAFORM: char       = '\u{f1062}'; // 󱁢
     const TEXT: char            = '\u{f15c}';  // 
     const TYPST: char           = '\u{f37f}';  // 
     const TMUX: char            = '\u{ebc8}';  // 
     const TOML: char            = '\u{e6b2}';  // 
+    const TODO: char            = '\u{f0ae}';  // 
     const TRANSLATION: char     = '\u{f05ca}'; // 󰗊
     const UNITY: char           = '\u{e721}';  // 
     const VECTOR: char          = '\u{f0559}'; // 󰕙
@@ -149,6 +158,7 @@ impl Icons {
     const VIM: char             = '\u{e7c5}';  // 
     const WRENCH: char          = '\u{f0ad}';  // 
     const XML: char             = '\u{f05c0}'; // 󰗀
+    const XORG: char            = '\u{f369}';  // 
     const YAML: char            = '\u{e6a8}';  // 
     const YARN: char            = '\u{e6a7}';  // 
 }
@@ -156,14 +166,18 @@ impl Icons {
 /// Mapping from full filenames to directory icon. This mapping should contain
 /// all the directories that have a custom icon.
 const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
+    ".build"              => Icons::FOLDER_WRENCH,  // 󱧼
     ".config"             => Icons::FOLDER_CONFIG,  // 
     ".exercism"           => Icons::FOLDER_EXERCISM,// 
     ".git"                => Icons::FOLDER_GIT,     // 
     ".github"             => Icons::FOLDER_GITHUB,  // 
     ".npm"                => Icons::FOLDER_NPM,     // 
     ".opam"               => Icons::FOLDER_OCAML,   // 
+    ".sources"            => Icons::FOLDER_CODE,    // 󰴉
     ".ssh"                => Icons::FOLDER_KEY,     // 󰢬
+    ".steam"              => Icons::STEAM,          // 
     ".Trash"              => '\u{f1f8}',            // 
+    "build"               => Icons::FOLDER_WRENCH,  // 󱧼
     "config"              => Icons::FOLDER_CONFIG,  // 
     "Contacts"            => '\u{f024c}',           // 󰉌
     "cron.d"              => Icons::FOLDER_CONFIG,  // 
@@ -176,6 +190,7 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
     "Downloads"           => '\u{f024d}',           // 󰉍
     "etc"                 => Icons::FOLDER_CONFIG,  // 
     "Favorites"           => '\u{f069d}',           // 󰚝
+    "friends"             => '\u{f024c}',           // 󰉌
     "hidden"              => Icons::FOLDER_HIDDEN,  // 󱞞
     "home"                => '\u{f10b5}',           // 󱂵
     "include"             => Icons::FOLDER_CONFIG,  // 
@@ -186,11 +201,15 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
     "npm_cache"           => Icons::FOLDER_NPM,     // 
     "pam.d"               => Icons::FOLDER_KEY,     // 󰢬
     "Pictures"            => '\u{f024f}',           // 󰉏
+    "sources"             => Icons::FOLDER_CODE,    // 󰴉
+    "src"                 => Icons::FOLDER_CODE,    // 󰴉
     "ssh"                 => Icons::FOLDER_KEY,     // 󰢬
+    "steam"               => Icons::STEAM,          // 
     "sudoers.d"           => Icons::FOLDER_KEY,     // 󰢬
+    "target"              => Icons::FOLDER_WRENCH,  // 󱧼
     "Videos"              => '\u{f03d}',            // 
     "xbps.d"              => Icons::FOLDER_CONFIG,  // 
-    "xorg.conf.d"         => Icons::FOLDER_CONFIG,  // 
+    "xorg.conf.d"         => Icons::XORG,           // 
     "cabal"               => Icons::LANG_HASKELL,   // 
 };
 
@@ -205,8 +224,8 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     ".bash_logout"        => Icons::SHELL,          // 󱆃
     ".bash_profile"       => Icons::SHELL,          // 󱆃
     ".CFUserTextEncoding" => Icons::OS_APPLE,       // 
-    ".clang-format"       => Icons::CONFIG,         // 
-    ".clang-tidy"         => Icons::CONFIG,         // 
+    ".clang-format"       => Icons::CONFIG,         // 󱁻
+    ".clang-tidy"         => Icons::CONFIG,         // 󱁻
     ".codespellrc"        => '\u{f04c6}',           // 󰓆
     ".condarc"            => '\u{e715}',            // 
     ".cshrc"              => Icons::SHELL,          // 󱆃
@@ -222,49 +241,49 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     ".eslintrc.yml"       => Icons::ESLINT,         // 
     ".gcloudignore"       => '\u{f11f6}',           // 󱇶
     ".fennelrc"           => Icons::LANG_FENNEL,    // 
-    ".gitattributes"      => Icons::GIT,            // 
-    ".git-blame-ignore-revs" => Icons::GIT,         // 
-    ".gitconfig"          => Icons::GIT,            // 
-    ".gitignore"          => Icons::GIT,            // 
-    ".gitignore_global"   => Icons::GIT,            // 
+    ".gitattributes"      => Icons::GIT,            // 󰊢
+    ".git-blame-ignore-revs" => Icons::GIT,         // 󰊢
+    ".gitconfig"          => Icons::GIT,            // 󰊢
+    ".gitignore"          => Icons::GIT,            // 󰊢
+    ".gitignore_global"   => Icons::GIT,            // 󰊢
     ".gitlab-ci.yml"      => '\u{f296}',            // 
-    ".gitmodules"         => Icons::GIT,            // 
+    ".gitmodules"         => Icons::GIT,            // 󰊢
     ".gtkrc-2.0"          => Icons::GTK,            // 
     ".gvimrc"             => Icons::VIM,            // 
-    ".htaccess"           => Icons::CONFIG,         // 
-    ".htpasswd"           => Icons::CONFIG,         // 
+    ".htaccess"           => Icons::CONFIG,         // 󱁻
+    ".htpasswd"           => Icons::CONFIG,         // 󱁻
     ".idea"               => Icons::INTELLIJ,       // 
     ".ideavimrc"          => Icons::VIM,            // 
-    ".inputrc"            => Icons::CONFIG,         // 
+    ".inputrc"            => Icons::CONFIG,         // 󱁻
     ".kshrc"              => Icons::SHELL,          // 󱆃
     ".login"              => Icons::SHELL,          // 󱆃
     ".logout"             => Icons::SHELL,          // 󱆃
-    ".luacheckrc"         => Icons::CONFIG,         // 
-    ".luaurc"             => Icons::CONFIG,         // 
+    ".luacheckrc"         => Icons::CONFIG,         // 󱁻
+    ".luaurc"             => Icons::CONFIG,         // 󱁻
     ".nanorc"             => '\u{e838}',            // 
     ".nuxtrc"             => '\u{f1106}',           // 󱄆
     ".ocamlinit"          => Icons::LANG_OCAML,     // 
-    ".mailmap"            => Icons::GIT,            // 
+    ".mailmap"            => Icons::GIT,            // 󰊢
     ".node_repl_history"  => Icons::NODEJS,         // 
     ".npmignore"          => Icons::NPM,            // 
     ".npmrc"              => Icons::NPM,            // 
     ".pre-commit-config.yaml" => '\u{f06e2}',       // 󰛢
     ".prettierignore"     => '\u{e6b4}',            // 
     ".prettierrc"         => '\u{e6b4}',            // 
-    ".parentlock"         => Icons::LOCK,           // 
+    ".parentlock"         => Icons::FILE_LOCK,      // 
     ".profile"            => Icons::SHELL,          // 󱆃
-    ".pylintrc"           => Icons::CONFIG,         // 
+    ".pylintrc"           => Icons::CONFIG,         // 󱁻
     ".python_history"     => Icons::LANG_PYTHON,    // 
     ".rustfmt.toml"       => Icons::LANG_RUST,      // 
     ".rvm"                => Icons::LANG_RUBY,      // 
     ".rvmrc"              => Icons::LANG_RUBY,      // 
-    ".SRCINFO"            => '\u{f303}',            // 
+    ".SRCINFO"            => Icons::ARCH_LINUX,     // 
     ".tcshrc"             => Icons::SHELL,          // 󱆃
     ".viminfo"            => Icons::VIM,            // 
     ".vimrc"              => Icons::VIM,            // 
-    ".Xauthority"         => Icons::CONFIG,         // 
-    ".xinitrc"            => Icons::CONFIG,         // 
-    ".Xresources"         => Icons::CONFIG,         // 
+    ".Xauthority"         => Icons::XORG,           // 
+    ".xinitrc"            => Icons::XORG,           // 
+    ".Xresources"         => Icons::XORG,           // 
     ".yarnrc"             => Icons::YARN,           // 
     ".zlogin"             => Icons::SHELL,          // 󱆃
     ".zlogout"            => Icons::SHELL,          // 󱆃
@@ -284,28 +303,32 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "Brewfile"            => '\u{f1116}',           // 󱄖
     "Brewfile.lock.json"  => '\u{f1116}',           // 󱄖
     "bspwmrc"             => '\u{f355}',            // 
+    "BUILD"               => Icons::WRENCH,         // 
     "build.gradle.kts"    => Icons::GRADLE,         // 
     "build.zig.zon"       => '\u{e6a9}',            // 
     "bun.lockb"           => '\u{e76f}',            // 
     "cantorrc"            => '\u{f373}',            // 
     "Cargo.lock"          => Icons::LANG_RUST,      // 
     "Cargo.toml"          => Icons::LANG_RUST,      // 
-    "CMakeLists.txt"      => '\u{e794}',            // 
+    "CHANGELOG"           => Icons::NEWS,           // 
+    "CHANGES"             => Icons::NEWS,           // 
+    "CHANGES.md"          => Icons::NEWS,           // 
+    "CMakeLists.txt"      => Icons::CMAKE,          // 
     "CODE_OF_CONDUCT"     => '\u{f4ae}',            // 
     "CODE_OF_CONDUCT.md"  => '\u{f4ae}',            // 
     "composer.json"       => Icons::LANG_PHP,       // 
     "composer.lock"       => Icons::LANG_PHP,       // 
-    "config"              => Icons::CONFIG,         // 
+    "config"              => Icons::CONFIG,         // 󱁻
     "config.ru"           => Icons::LANG_RUBY,      // 
-    "config.status"       => Icons::CONFIG,         // 
+    "config.status"       => Icons::CONFIG,         // 󱁻
     "configure"           => Icons::WRENCH,         // 
-    "configure.ac"        => Icons::CONFIG,         // 
-    "configure.in"        => Icons::CONFIG,         // 
+    "configure.ac"        => Icons::CONFIG,         // 󱁻
+    "configure.in"        => Icons::CONFIG,         // 󱁻
     "constraints.txt"     => Icons::LANG_PYTHON,    // 
-    "COPYING"             => Icons::LICENSE,        // 
-    "COPYRIGHT"           => Icons::LICENSE,        // 
-    "crontab"             => Icons::CONFIG,         // 
-    "crypttab"            => Icons::CONFIG,         // 
+    "COPYING"             => Icons::LICENSE,        // 
+    "COPYRIGHT"           => Icons::LICENSE,        // 
+    "crontab"             => Icons::CONFIG,         // 󱁻
+    "crypttab"            => Icons::CONFIG,         // 󱁻
     "csh.cshrc"           => Icons::SHELL,          // 󱆃
     "csh.login"           => Icons::SHELL,          // 󱆃
     "csh.logout"          => Icons::SHELL,          // 󱆃
@@ -318,7 +341,8 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "dune-project"        => Icons::WRENCH,         // 
     "Earthfile"           => '\u{f0ac}',            // 
     "COMMIT_EDITMSG"      => Icons::GIT,            // 
-    "environment"         => Icons::CONFIG,         // 
+    "environment"         => Icons::CONFIG,         // 󱁻
+    "env"                 => Icons::CONFIG,         // 󱁻
     "favicon.ico"         => '\u{e623}',            // 
     "flake.lock"          => '\u{f313}',            // 
     "fennelrc"            => Icons::LANG_FENNEL,    // 
@@ -346,7 +370,7 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "gulpfile.js"         => Icons::GULP,           // 
     "gulpfile.ls"         => Icons::GULP,           // 
     "heroku.yml"          => '\u{e77b}',            // 
-    "hostname"            => Icons::CONFIG,         // 
+    "hostname"            => Icons::CONFIG,         // 󱁻
     "hypridle.conf"       => '\u{f359}',            // 
     "hyprland.conf"       => '\u{f359}',            // 
     "hyprlock.conf"       => '\u{f359}',            // 
@@ -359,8 +383,8 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "id_ed25519"          => Icons::PRIVATE_KEY,    // 󰌆
     "id_ed25519_sk"       => Icons::PRIVATE_KEY,    // 󰌆
     "id_rsa"              => Icons::PRIVATE_KEY,    // 󰌆
-    "index.theme"     => '\u{ee72}',           // 
-    "inputrc"             => Icons::CONFIG,         // 
+    "index.theme"         => '\u{ee72}',            // 
+    "inputrc"             => Icons::CONFIG,         // 󱁻
     "Jenkinsfile"         => '\u{e66e}',            // 
     "jsconfig.json"       => Icons::LANG_JAVASCRIPT,// 
     "Justfile"            => Icons::WRENCH,         // 
@@ -372,18 +396,18 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "known_hosts"         => '\u{f08c0}',           // 󰣀
     "kritadisplayrc"      => Icons::KRITA,          // 
     "kritarc"             => Icons::KRITA,          // 
-    "LICENCE"             => Icons::LICENSE,        // 
-    "LICENCE.md"          => Icons::LICENSE,        // 
-    "LICENCE.txt"         => Icons::LICENSE,        // 
-    "LICENSE"             => Icons::LICENSE,        // 
-    "LICENSE-APACHE"      => Icons::LICENSE,        // 
-    "LICENSE-MIT"         => Icons::LICENSE,        // 
-    "LICENSE.md"          => Icons::LICENSE,        // 
-    "LICENSE.txt"         => Icons::LICENSE,        // 
+    "LICENCE"             => Icons::LICENSE,        // 
+    "LICENCE.md"          => Icons::LICENSE,        // 
+    "LICENCE.txt"         => Icons::LICENSE,        // 
+    "LICENSE"             => Icons::LICENSE,        // 
+    "LICENSE-APACHE"      => Icons::LICENSE,        // 
+    "LICENSE-MIT"         => Icons::LICENSE,        // 
+    "LICENSE.md"          => Icons::LICENSE,        // 
+    "LICENSE.txt"         => Icons::LICENSE,        // 
     "localized"           => Icons::OS_APPLE,       // 
     "localtime"           => Icons::CLOCK,          // 
-    "lock"                => Icons::LOCK,           // 
-    "LOCK"                => Icons::LOCK,           // 
+    "lock"                => Icons::FILE_LOCK,      // 
+    "LOCK"                => Icons::FILE_LOCK,      // 
     "log"                 => Icons::LOG,            // 
     "LOG"                 => Icons::LOG,            // 
     "lxde-rc.xml"         => '\u{f363}',            // 
@@ -397,13 +421,14 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "MANIFEST.in"         => Icons::LANG_PYTHON,    // 
     "mix.lock"            => Icons::LANG_ELIXIR,    // 
     "mpv.conf"            => '\u{f36e}',            // 
+    "NEWS"                => Icons::NEWS,           // 
     "npm-shrinkwrap.json" => Icons::NPM,            // 
     "npmrc"               => Icons::NPM,            // 
     "package-lock.json"   => Icons::NPM,            // 
     "package.json"        => Icons::NPM,            // 
     "passwd"              => Icons::LOCK,           // 
     "php.ini"             => Icons::LANG_PHP,       // 
-    "PKGBUILD"            => '\u{f303}',            // 
+    "PKGBUILD"            => Icons::ARCH_LINUX,     // 
     "platformio.ini"      => '\u{e682}',            // 
     "pom.xml"             => '\u{e674}',            // 
     "Procfile"            => '\u{e77b}',            // 
@@ -428,13 +453,16 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     "SECURITY.md"         => '\u{f0483}',           // 󰒃
     "settings.gradle.kts" => Icons::GRADLE,         // 
     "shadow"              => Icons::LOCK,           // 
-    "shells"              => Icons::CONFIG,         // 
+    "shells"              => Icons::CONFIG,         // 󱁻
     "sudoers"             => Icons::LOCK,           // 
-    "sxhkdrc"             => Icons::CONFIG,         // 
+    "sxhkdrc"             => Icons::CONFIG,         // 󱁻
     "sym-lib-table"       => Icons::KICAD,          // 
     "timezone"            => Icons::CLOCK,          // 
     "tmux.conf"           => Icons::TMUX,           // 
     "tmux.conf.local"     => Icons::TMUX,           // 
+    "TASKS"               => Icons::TODO,           // 
+    "TDL"                 => Icons::TODO,           // 
+    "TODO"                => Icons::TODO,           // 
     "tsconfig.json"       => Icons::LANG_TYPESCRIPT,// 
     "Vagrantfile"         => '\u{2371}',            // ⍱
     "vlcrc"               => '\u{f057c}',           // 󰕼
@@ -465,7 +493,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "7z"             => Icons::COMPRESSED,       // 
     "a"              => Icons::OS_LINUX,         // 
     "aac"            => Icons::AUDIO,            // 
-    "acf"            => '\u{f1b6}',              // 
+    "acf"            => Icons::STEAM,            // 
     "age"            => Icons::SHIELD_LOCK,      // 󰦝
     "ai"             => '\u{e7b4}',              // 
     "aif"            => Icons::AUDIO,            // 
@@ -516,19 +544,19 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "cbz"            => Icons::IMAGE,            // 
     "cc"             => Icons::LANG_CPP,         // 
     "cert"           => Icons::GIST_SECRET,      // 
-    "cfg"            => Icons::CONFIG,           // 
+    "cfg"            => Icons::CONFIG,           // 󱁻
     "cjs"            => Icons::LANG_JAVASCRIPT,  // 
     "class"          => Icons::LANG_JAVA,        // 
     "clj"            => '\u{e768}',              // 
     "cljs"           => '\u{e76a}',              // 
     "cls"            => Icons::LANG_TEX,         // 
-    "cmake"          => '\u{e794}',              // 
+    "cmake"          => Icons::CMAKE,            // 
     "cmd"            => Icons::OS_WINDOWS,       // 
     "coffee"         => '\u{f0f4}',              // 
     "com"            => '\u{e629}',              // 
     "conda"          => '\u{e715}',              // 
-    "conf"           => Icons::CONFIG,           // 
-    "config"         => Icons::CONFIG,           // 
+    "conf"           => Icons::CONFIG,           // 󱁻
+    "config"         => Icons::CONFIG,           // 󱁻
     "cow"            => '\u{f019a}',             // 󰆚
     "cp"             => Icons::LANG_CPP,         // 
     "cpio"           => Icons::COMPRESSED,       // 
@@ -559,7 +587,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "diff"           => Icons::DIFF,             // 
     "djv"            => Icons::DOCUMENT,         // 
     "djvu"           => Icons::DOCUMENT,         // 
-    "dll"            => Icons::LIBRARY,          // 
+    "dll"            => Icons::OS_WINDOWS,       // 
     "dmg"            => Icons::DISK_IMAGE,       // 
     "doc"            => Icons::DOCUMENT,         // 
     "docm"           => Icons::DOCUMENT,         // 
@@ -640,7 +668,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "gemspec"        => Icons::LANG_RUBY,        // 
     "gform"          => '\u{f298}',              // 
     "gif"            => Icons::IMAGE,            // 
-    "git"            => Icons::GIT,              // 
+    "git"            => Icons::GIT,              // 󰊢
     "gleam"          => Icons::LANG_GLEAM,       // 󰦥
     "gm1"            => Icons::EDA_PCB,          // 
     "gml"            => Icons::EDA_PCB,          // 
@@ -694,7 +722,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "img"            => Icons::DISK_IMAGE,       // 
     "iml"            => Icons::INTELLIJ,         // 
     "info"           => Icons::INFO,             // 
-    "ini"            => Icons::CONFIG,           // 
+    "ini"            => Icons::CONFIG,           // 󱁻
     "inl"            => Icons::LANG_C,           // 
     "ipynb"          => '\u{e80f}',              // 
     "ino"            => Icons::LANG_ARDUINO,     // 
@@ -753,10 +781,10 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "lff"            => Icons::FONT,             // 
     "lhs"            => Icons::LANG_HASKELL,     // 
     "lib"            => Icons::LIBRARY,          // 
-    "license"        => Icons::LICENSE,          // 
+    "license"        => Icons::LICENSE,          // 
     "lisp"           => '\u{f0172}',             // 󰅲
     "localized"      => Icons::OS_APPLE,         // 
-    "lock"           => Icons::LOCK,             // 
+    "lock"           => Icons::FILE_LOCK,        // 
     "log"            => Icons::LOG,              // 
     "lpp"            => Icons::EDA_PCB,          // 
     "lrc"            => Icons::SUBTITLE,         // 󰨖
@@ -984,11 +1012,11 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "tape"           => '\u{F0A1B}',             // 󰨛
     "tar"            => Icons::COMPRESSED,       // 
     "taz"            => Icons::COMPRESSED,       // 
-    "tbc"            => '\u{f06d3}',             // 󰛓
+    "tbc"            => Icons::TCL,              // 󰛓
     "tbz"            => Icons::COMPRESSED,       // 
     "tbz2"           => Icons::COMPRESSED,       // 
     "tc"             => Icons::DISK_IMAGE,       // 
-    "tcl"            => '\u{f06d3}',             // 󰛓
+    "tcl"            => Icons::TCL,              // 󰛓
     "tex"            => Icons::LANG_TEX,         // 
     "tf"             => Icons::TERRAFORM,        // 󱁢
     "tfstate"        => Icons::TERRAFORM,        // 󱁢
@@ -997,10 +1025,11 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "tif"            => Icons::IMAGE,            // 
     "tiff"           => Icons::IMAGE,            // 
     "tlz"            => Icons::COMPRESSED,       // 
-    "tml"            => Icons::CONFIG,           // 
+    "tml"            => Icons::CONFIG,           // 󱁻
     "tmux"           => Icons::TMUX,             // 
     "toml"           => Icons::TOML,             // 
     "torrent"        => '\u{e275}',              // 
+    "tpl"            => Icons::PLAYLIST,         // 󰲹
     "tres"           => Icons::GODOT,            // 
     "ts"             => Icons::LANG_TYPESCRIPT,  // 
     "tscn"           => Icons::GODOT,            // 
@@ -1020,6 +1049,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "v"              => Icons::LANG_V,           // 
     "vala"           => '\u{e8d1}',              // 
     "vdi"            => Icons::DISK_IMAGE,       // 
+    "vdf"            => Icons::STEAM,            // 
     "vhd"            => Icons::DISK_IMAGE,       // 
     "vhs"            => '\u{F0A1B}',             // 󰨛
     "vi"             => '\u{e81e}',              // 
